core/src/commonMain/kotlin/com/powersync/attachments/sync/SyncingService.kt

```diff
@@ -0,0 +1,278 @@
+                    val watchJob =
+                        launch {
+                            merge(
+                                // Handles manual triggers for sync events
+                                syncTriggerFlow.asSharedFlow(),
+                                // Triggers the sync process whenever an underlaying change to the
+                                // attachments table happens
+                                attachmentsService
+                                    .watchActiveAttachments(),
+                            )
+                                // We only use these flows to trigger the process. We can skip multiple invocations
+                                // while we are processing. We will always process on the trailing edge.
+                                // This buffer operation should automatically be applied to all merged sources.
+                                .buffer(1, onBufferOverflow = BufferOverflow.DROP_OLDEST)
+                                .throttle(syncThrottle.inWholeMilliseconds)
+                                .collect {
+                                    attachmentsService.withLock { context ->
+                                        /**
+                                         * Gets and performs the operations for active attachments which are
+                                         * pending download, upload, or delete.
+                                         */
+                                        try {
+                                            val attachments = context.getActiveAttachments()
+                                            // Performs pending operations and updates attachment states
+                                            handleSync(attachments, context)
+
+                                            // Cleanup archived attachments
+                                            deleteArchivedAttachments(context)
+                                        } catch (ex: Exception) {
+                                            if (ex is CancellationException) {
+                                                throw ex
+                                            }
+                                            // Rare exceptions caught here will be swallowed and retried on the
+                                            // next tick.
+                                            logger.e("Caught exception when processing attachments $ex")
+                                        }
+                                    }
+                                }
+                        }
+
+                    val periodicJob =
+                        launch {
+                            logger.i("Periodically syncing attachments")
+                            syncTriggerFlow.emit(Unit)
+                            delay(period)
+                        }
+
+                    watchJob.join()
+                    periodicJob.join()
```

Comment:
Mainly out of my own interest, but are these necessary? It was my understanding that the parent job would be active as long as it has an active child job too.

Reply:
True. This was not necessary. Removed it.